projects/laji/src/app/shared-modules/select/select-subcategories/select-subcategories.component.ts

```diff
@@ -248,7 +248,7 @@ export class SelectSubcategoriesComponent implements OnChanges {
   }
 
   private rebuiltParamSubCategory(urlString) {
-    const rebuilt = urlString.slice(0, -1).split(';');
+    const rebuilt = urlString.charAt(urlString.length - 1) === ';' ? urlString.slice(0, -1).split(';') : urlString.split(';');
```

Comment:
This would be a bit cleaner:
```
const rebuilt = urlString.split(';').filter(s => s.length > 0);
```

Reply:
Also the function could be renamed to something like `arrFromUrlParams` and have corresponding typing.

Reply:
Also, imo utility functions like this should be defined outside of the class (or as a static function in pure OOP languages). The function doesn't require a class instance to be defined, so being inside a class just adds extra mental overhead/context.

Reply:
@MLumme is on vacation so I took over on making the code changes & approved

Reply:
The class has also other utils as class methods, so for consistency I didn't want to introduce a "new scope", so I put the function inside the class method that needs the util